### DISCUSSION TITLE
Would there be a benefit to Owned<T>.Instance throwing ObjectDisposedException?

### DISCUSSION THREAD

**Discussion by jnm2** [YairHalberstadt/stronginject#161] [Ideas]
It's probably too late until v2 if ever, but I wondered what your thoughts were. Is accessing `Instance` potentially _not_ a mistake after letting the `Owned<T>` be disposed?

**Comment by YairHalberstadt** [comment]
On the one hand, yes it is a mistake. On the other hand is super easy to make this mistake anyway, by keeping Instance around rather than accessing it each time.

Finally it will require not allowing a null action, or introducing an extra bit of state.

**Reply by jnm2** [reply]
(My implementation didn't require extra state. It achieved this by coalescing the constructor parameter with a cached EmptyAction delegate.)

I think that's a pretty good point though, it not being a sufficient check to prevent much. Also, I ran into one case where I did need to use the value after disposal and the use was legitimate. It was to use the value as an opaque object reference as a lookup key. So it's neither a sufficient check nor always a helpful one.
